[E0735389] iP

E0735389 opened the pull request:
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ *SUPER FAST* to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

------

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Duke {
    public static void main(String[] args) {
        Commands.registerCommands();
        Duke duke = new Duke();
        duke.run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

E0735389 committed:
Level 0: Rename, Greet, Exit

E0735389 committed:
Level 1

E0735389 committed:
Level 2: Add, List

E0735389 committed:
Level 3: mark as done

E0735389 committed:
Level 4: ToDo, Event, Deadline

E0735389 committed:
Add test

E0735389 committed:
Level-6: Delete tasks

E0735389 committed:
Slight update to README

E0735389 committed:
Level 7. Save

E0735389 committed:
Merge branch 'branch-Level-7'

E0735389 committed:
Implement date and time support

E0735389 committed:
Implement clear command to make test robust

E0735389 committed:
Merge branch 'branch-Level-8'

E0735389 committed:
Refactor, extract Ui class

E0735389 committed:
Extract class TaskList

E0735389 committed:
Organize classes into packages

hjungwoo01 commented on src/main/java/ByeCommand.java:
```diff
@@ -0,0 +1,11 @@
+
+class ByeCommand extends Command {
+    @Override
+    public void execute(TaskList tasks, boolean silent) {
```
Consider using more descriptive names for your boolean variables and methods to enhance code readability. I suggest renaming 'silent' to 'isSilent' to make its purpose clearer.

hjungwoo01 commented on src/main/java/Commands.java:
```diff
@@ -0,0 +1,186 @@
+package duke;
+
+import java.util.regex.Pattern;
```
It would be beneficial to maintain a consistent import order for better code organization. I recommend placing the 'import java.util.regex.Matcher' statement above the 'import java.util.regex.Pattern' statement to follow the convention of arranging import statements in lexical order.

hjungwoo01 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,33 @@
+import java.util.HashMap;
+
+public class Parser {
+    private static HashMap<String, CommandBuilder> registeredCommands = new HashMap<>();
```
Consider initializing 'registeredCommands' using the 'Map<>' interface instead of 'HashMap<>' for improved flexibility and adherence to coding to interfaces best practices. So something like Map<String, CommandBuilder> registeredCommands = new HashMap<>()

hjungwoo01 reviewed (commented):
Overall, I found your code to be well-written. It was generally easy to read, and I didn't notice any issues. Great job overall!

E0735389 committed:
Actually organize things into packages

E0735389 committed:
Merge branch 'add-gradle-support'

E0735389 committed:
Fix main class package for Gradle

E0735389 committed:
Add tests

E0735389 committed:
Add JavaDoc documentation

E0735389 committed:
Implement level 9 (find) and fix some bugs

E0735389 committed:
Merge branch 'branch-A-JavaDoc'

E0735389 committed:
Merge branch 'branch-Level-9'

E0735389 committed:
Modify build.gradle for Checkstyle

E0735389 committed:
Install checkstyle

E0735389 committed:
Merge branch 'branch-A-CheckStyle'

E0735389 committed:
Implement GUI

E0735389 committed:
Remove text-ui-test

E0735389 committed:
Merge branch 'branch-Level-10'

E0735389 committed:
Add assertions to check important assumptions

The code does not use assertions to check for some
invariants in the code.

This makes it slightly harder to understand what are the conditions
satisfied by the program state at each point of the execution.

This commit adds assertions at various points in the code.

E0735389 committed:
Merge branch 'branch-A-Assertions'

E0735389 committed:
Improve the code quality

Many methods that overrides a method or implements an abstract method
in the parent class/interface does not have the @Override annotation.

This is bad coding practice, and does not allow the compiler to
raise warnings about incorrect overriding.

This commit changes the situation by adding @Override at
various places.

E0735389 committed:
Merge pull request #2 from E0735389/branch-A-CodeQuality

Improve the code quality

E0735389 committed:
Check for clashing tasks and related improvements

Two clashing events can be added at the same time.

This is bad because the user may accidentally enter the wrong
event starting/ending time and not notice it.

Let's,
* check for clashing tasks before adding it
* in the meantime, make sure that the starting time of an event is
  before the ending time

E0735389 committed:
Merge pull request #3 from E0735389/branch-B-DetectAnomalies

Check for clashing tasks and related improvements

E0735389 committed:
Set window title and delete redundant Main class

E0735389 committed:
Add representative screenshot and user guide

E0735389 committed:
Improve code quality and fix unit tests

E0735389 committed:
Fix style errors, add JavaDoc to several methods

E0735389 committed:
Add a test for AddDeadlineCommand
